data/ha/hawk_test.py

```diff
@@ -0,0 +1,829 @@
+              myprimitive)
+        set_test_status('remove_primitive', 'failed')
+        RETURN_VALUE = 26
+
+### MAIN
+
+# Command line argument parsing
+parser = argparse.ArgumentParser(description='HAWK GUI interface Selenium test')
+parser.add_argument('-b', '--browser', type=str,
+                    help='Browser to use in the test. Can be: firefox, chrome, chromium')
+parser.add_argument('-H', '--host', type=str, default='localhost',
+                    help='Host or IP address where HAWK is running')
+parser.add_argument('-P', '--port', type=str, default='7630',
+                    help='TCP port where HAWK is running')
+parser.add_argument('-p', '--prefix', type=str, default='',
+                    help='Prefix to add to Resources created during the test')
+parser.add_argument('-t', '--test-version', type=str, default='',
+                    help='Test version. Ex: 12-SP3, 12-SP4, 15, 15-SP1')
+parser.add_argument('-s', '--secret', type=str, default='',
+                    help='root SSH Password of the HAWK node')
+parser.add_argument('-r', '--results', type=str, default='',
+                    help='Generate hawk_test.results file')
+args = parser.parse_args()
+if not args.browser:
+    print('ERROR: must specify a browser with --browser')
+    parser.print_help()
+    quit(3)
+
+# Set URL to use
+mainlink = 'https://' + args.host.lower() + ':' + args.port
+
+# Create appropiate driver
+if args.browser.lower() in ['chrome', 'chromium']:
+    browser = webdriver.Chrome()
+    timeout_scale = 1
+elif args.browser.lower() == 'firefox':
+    profile = webdriver.FirefoxProfile()
+    profile.accept_untrusted_certs = True
+    profile.assume_untrusted_cert_issuer = True
+    browser = webdriver.Firefox(firefox_profile=profile)
+    timeout_scale = 2.5
+else:
+    print('ERROR: --browser must be firefox, chrome or chromium')
+    quit(3)
+browser.maximize_window()
+
+# Establish SSH connection to verify status only if SSH password was supplied
+if args.secret:
+    ssh = paramiko.SSHClient()
+    ssh.load_system_host_keys()
+    ssh.set_missing_host_key_policy(paramiko.WarningPolicy)
+    ssh.connect(hostname=args.host.lower(), username="root", password=args.secret)
+
+# Initialize result set
+initialize_result_set(args.test_version.lower(), args.browser.lower())
+
+# Resources to create
+if args.prefix and not re.match(r"^\w+$", args.prefix.lower()):
+    print("ERROR: Prefix must contain only numbers and letters. Ignoring")
+    args.prefix = ''
+mycluster = args.prefix.lower() + 'Anderes'
+myprimitive = args.prefix.lower() + 'cool_primitive'
+myclone = args.prefix.lower() + 'cool_clone'
+mygroup = args.prefix.lower() + 'cool_group'
+
+# Variables for texts that change between chrome and firefox drivers
+troubleshoot = link_by_browser(browser, 'troubleshooting')
+monitoring = link_by_browser(browser, 'monitoring')
+
+# Tests to perform
+browser.get(mainlink)
```

Comment:
below is some dirty mix of tests functions called and I suspect depend on each other and even some action happening in between. 
Each test case should be started with new instance of web driver which will give you clean cache and no side-effects of other tests . Otherwise you will have hard times to maintain all this , also you will have hard times to be confident that if test C failed you really have problem  with feature C and not that test A leave some crap behind which cause feature C not to work .

Reply:
Wouldn't this mean that a new session (including logging in to hawk) is started with each new instance of webdriver? It is a good idea, and I figure it will help with some tests particularly when using this same script to test hawk versions in 12-SP*, but I worry about how much more time this would add to the test. Currently it's taking close to 7 minutes (which is why I'm giving the `ha/hawk_gui` test module 15 minutes to work). Will look into it though, as it'll be easier to maintain. Thanks!

Reply:
every solution has it's pros and cons. Having new instance of webdriver would slow you down. But I don't think that it will slow you down dramatically. Also you will get huge benefits - additionally to what I already mention you can feel more relaxed about failures in the test. Each your test would  be capable to fail only itself but not whole test suite